File: backend/internal/models/experiencia.go
package models

// ExperienciaProfissional representa uma experiência profissional
type ExperienciaProfissional struct {
	ID         int      `json:"id,omitempty" db:"id"`
	Cargo      string   `json:"cargo" db:"cargo"`
	Empresa    string   `json:"empresa" db:"empresa"`
	Localizacao string  `json:"localizacao" db:"localizacao"`
	Periodo    string   `json:"periodo" db:"periodo"`
	Ordem      int      `json:"ordem,omitempty" db:"ordem"`
	Atividades []string `json:"atividades"`
}

// CreateExperienciaRequest representa a requisição para criar uma experiência
type CreateExperienciaRequest struct {
	Cargo      string   `json:"cargo"`
	Empresa    string   `json:"empresa"`
	Localizacao string  `json:"localizacao"`
	Periodo    string   `json:"periodo"`
	Ordem      int      `json:"ordem"`
	Atividades []string `json:"atividades"`
}

// UpdateExperienciaRequest representa a requisição para atualizar uma experiência
type UpdateExperienciaRequest struct {
	Cargo      string   `json:"cargo"`
	Empresa    string   `json:"empresa"`
	Localizacao string  `json:"localizacao"`
	Periodo    string   `json:"periodo"`
	Ordem      int      `json:"ordem"`
	Atividades []string `json:"atividades"`
}

File: frontend/src/types/Experiencia.ts
export interface ExperienciaProfissional {
  id?: number
  cargo: string
  empresa: string
  localizacao: string
  periodo: string
  ordem?: number
  atividades: string[]
}

export interface CreateExperienciaRequest {
  cargo: string
  empresa: string
  localizacao: string
  periodo: string
  ordem: number
  atividades: string[]
}

export interface UpdateExperienciaRequest {
  cargo: string
  empresa: string
  localizacao: string
  periodo: string
  ordem: number
  atividades: string[]
}

